### PR TITLE
simpler and more robust deletion error handling

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?177171

a few changes bundled together
- don't have one failing case break the whole process
- only retry cases that fail
- trap `ResourceConflict` in addition to weird form deletions
- lower delay

### CODE REVIEW

File: corehq/apps/users/tasks.py [hunk 1]
```diff
@@ -65,16 +63,23 @@ def tag_forms_as_deleted_rebuild_associated_cases(self, formlist, deletion_id,
         cases_to_rebuild.update(get_case_ids_from_form(form))
     XFormInstance.get_db().bulk_save(formlist)
 
+    for case in cases_to_rebuild - deleted_cases:
+        _rebuild_case_with_retries.delay(case)
+
+@task(bind=True, rate_limit=2, queue='background_queue', ignore_result=True,
```
Comment (dimagimon) [inline corehq/apps/users/tasks.py]: E302 expected 2 blank lines, found 1

**Comment by czue** [comment]
@biyeun recently was working on this and @snopoke was looking over my shoulder for a bit. cc @emord

File: corehq/apps/users/tasks.py [hunk 2]
```diff
@@ -65,16 +63,24 @@ def tag_forms_as_deleted_rebuild_associated_cases(self, formlist, deletion_id,
         cases_to_rebuild.update(get_case_ids_from_form(form))
     XFormInstance.get_db().bulk_save(formlist)
 
+    for case in cases_to_rebuild - deleted_cases:
+        _rebuild_case_with_retries.delay(case)
```
Comment (snopoke) [inline corehq/apps/users/tasks.py]: Do we want a single task per case or should we use something like celery map / chunks: http://celery.readthedocs.org/en/latest/userguide/canvas.html#chunks

Comment (czue) [inline corehq/apps/users/tasks.py]: cool - yeah i decided it didn't really matter

**Pull request merged**